### ISSUE TITLE
Change promt to be more useful

### ISSUE THREAD

**Issue by tryoxiss** [Kubulambula/Godot-GDShell#24]
I finally got it set up in a project and I was wondering what that was for. I know thats default in bash, `user@computer`, but here it dosen't seem to serve a pourpose other than to mimic BASH. I would say the seccond variable can be used for the current scene but I don't know what the first one could really be fore. (e.g., `main_menu$ [enter command]`).

Personally I don't really see a point in having constants in the promt more than necesary, especially not when thier only pourpose is to mimic bash. It can -- and does -- have a familliar feel without that.

**Comment by Kubulambula** [comment]
It really serves no purpose other than to "look cool". We can definitely give it a purpose and maybe make it accessible to be modified by commands?

> I would say the seccond variable can be used for the current scene but I don't know what the first one could really be fore. (e.g., `main_menu$ [enter command]`).

What about `scene_name@project_name`? If I remember correctly, currently it does show `gdshell@project_name`.

**Comment by tryoxiss** [comment]
Allowing it to be modified by commands (especially autorun which runs at the start) would be really good!

That would work but the project name is not really something needed most of the time, and theres gdfetch for that when it is. And it dosen't seem to be the projects name. (see screenshot)

![image](https://github.com/Kubulambula/Godot-GDShell/assets/94419893/79c44855-8e91-4624-af2c-600180840217)

If it is made customiseable some things that could be useful is

- Current scene
- Current memory useage
- Allocated threeads
- is it safe or unsafe mode ($ could be safe, # could be unsafe maybe?)
- Arbitrary strings or code variables
- godot version
- build (editor/export)
- Is the scene "ready" (ready/loading)

Having a few promt presets may also be good. Such as `scene [mem_used/mem_alloc] $ `

**Comment by Kubulambula** [comment]
I don't really know about having monitors hooked up to it (like the memory usage) as there's a separate command for it already.
But Godot version and separate symbols to show if the GDShell is in safe or unsafe mode are great ideas.

**Comment by CollCaz** [comment]
I think the prompt should display info on where you are, just like in the shell
Maybe it could show you which scene is currently running?

**Comment by tryoxiss** [comment]
> I don't really know about having monitors hooked up to it (like the memory usage) as there's a separate command for it already. But Godot version and separate symbols to show if the GDShell is in safe or unsafe mode are great ideas.

Yeah I was just dropping any idea I thought of, looking at it again a seperate command is just better.
